File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<double> seen;
        
        for (int num : arr){
            if (seen.count(num / 2.0) || seen.count(num * 2))
                return true;
            seen.insert(num);
        }
        
        return false;
    }
};

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        seen = set()
        
        for num in arr:
            # Check if 2 * num or num / 2 exists in the set
            if 2 * num in seen or (num % 2 == 0 and (num >> 1) in seen):
                return True
            
            # Add the current number to the set
            seen.add(num)
            
        # No valid pair found
        return False